debian/patches/fast-nocloud-network.patch

```diff
@@ -0,0 +1,64 @@
+Description: Minimize network-config wait
+Minimize possible wait time for a NoCloud file that likely
+doesn't exist on existing series.
+Bug: https://github.com/canonical/cloud-init/issues/5753
+
+Author: Brett Holman <brett.holman@canonical.com>
+Last-Update: 2024-11-13
+
+--- a/cloudinit/util.py
++++ b/cloudinit/util.py
+@@ -1021,16 +1021,6 @@ def read_seeded(base="", ext="", timeout
+         vd_url = "%s%s%s" % (base, "vendor-data", ext)
+         md_url = "%s%s%s" % (base, "meta-data", ext)
+         network_url = "%s%s%s" % (base, "network-config", ext)
+-    network = None
+-    try:
+-        network_resp = url_helper.read_file_or_url(
+-            network_url, timeout=timeout, retries=retries
+-        )
+-    except url_helper.UrlError as e:
+-        LOG.debug("No network config provided: %s", e)
+-    else:
+-        if network_resp.ok():
+-            network = load_yaml(network_resp.contents)
+     md_resp = url_helper.read_file_or_url(
+         md_url, timeout=timeout, retries=retries
+     )
+@@ -1058,6 +1048,18 @@ def read_seeded(base="", ext="", timeout
+         else:
+             LOG.debug("Error in vendor-data response")
+
++    network = None
++    if md or ud or vd:
++        try:
++            network_resp = url_helper.read_file_or_url(
++                network_url, timeout=timeout, retries=0
```

Comment:
This is going to add a 5s timeout to every NoCloud server which doesn't provide a network-config. That sounds like change in a change in behavior that would negatively affect some users.

Reply:
Possible, but I don't think that will be true in most cases. Any reasonable server will return a 404 immediately, and with retries=0, that should finish fairly quick. In the case of a firewall swallowing packets, the previous calls should fail first, preventing this one from running.

Reply:
> Possible, but I don't think that will be true in most cases.

Sure, agreed.

> Any reasonable server will return a 404 immediately, and with retries=0, that should finish fairly quick.

I don't disagree - for a reasonable server this should probably be just fine. I'm just concerned about user infrastructure that doesn't behave reasonably. Is "anything broken by this change is unreasonable" reasonable justification for shipping a change in behavior?